### PR TITLE
Patch ui tooltips 419b

### LINKED ISSUE #419: polish tooltip looks and fix orphan behaviour. (was Implement Patch UI tooltip styles)
patch tooltips

You already have them basically going @gmarinov just needs minor polish for size & style.
Sketch file reference Patch UI tooltips 151015

### PR DESCRIPTION
**Pull request by gmarinov**
fixes #419

### CODE REVIEW

File: browser/scripts/node-ui.js [hunk 1]
```diff
@@ -161,7 +163,7 @@ NodeUI = function(parent_node, x, y, z) {
 
 	if (parent_node.plugin.desc) {
 		$header.attr('alt', '' + parent_node.uid);
-		$header.hover(E2.app.onShowTooltip.bind(E2.app), E2.app.onHideTooltip.bind(E2.app));
+		$header.hover(NodeUI.onShowTooltip.bind($header), NodeUI.onHideTooltip.bind($header));
```
Comment (kschzt) [inline browser/scripts/node-ui.js]: shouldn't this be `this.onShowTooltip()` instead of a static method?

File: browser/scripts/node-ui.js [hunk 2]
```diff
@@ -186,6 +188,17 @@ NodeUI = function(parent_node, x, y, z) {
 
 }
 
+//NodeUI.prototype = Object.create(EventEmitter.prototype);
+
+NodeUI.prototype.destroy = function() {
+	jQuery('div.popover').remove();	// clean up any tooltips
+	[this.input_col, this.output_col, this.inline_in,
+		this.inline_out, this.header, this.content,
+		this.plugin_container, this.plugin_ui].forEach(function(j){ if (typeof j === 'function') j.remove(); j={}; });
```
Comment (kschzt) [inline browser/scripts/node-ui.js]: readability. please put the .forEach on a new line and expand the function

**Comment by kschzt** [comment]
otherwise lgtm

**Comment by gmarinov** [comment]
@kschzt please take a look

File: browser/scripts/node-ui.js [hunk 3]
```diff
@@ -270,41 +338,37 @@ NodeUI.prototype.canDisplayInputInHeader = function() {
...
 	var myCategory = this.getNodeCategory();
-	can &= (uiNodeCategoriesThatNormallyDisplayOutputInHeader.indexOf(myCategory) > -1);
-	can &= (exceptPlugins.indexOf(this.parent_node.plugin.id) === -1);
+	can = can && (uiNodeCategoriesThatNormallyDisplayOutputInHeader.indexOf(myCategory) > -1);
+	can = can && (exceptPlugins.indexOf(this.parent_node.plugin.id) === -1);
 
 	return can;
 };
 
 NodeUI.prototype.canDisplayInline = function() {
```
Comment (kschzt) [inline browser/scripts/node-ui.js]: this really calls out for a unit test.

File: browser/scripts/node-ui.js [hunk 4]
```diff
@@ -501,7 +703,7 @@ NodeUI.makeSpriteSVG = function(xlink, className) {
  * @returns jQuery
  */
 NodeUI.makeSpriteSVGButton = function($svg, alt_text, $have_button) {
```
Comment (kschzt) [inline browser/scripts/node-ui.js]: camelCase

**Comment by kschzt** [comment]
lgtm, and if you add a unit test, looks awesome to me

File: browser/scripts/node-ui.js (hunk 3 above)
Comment (gmarinov) [inline browser/scripts/node-ui.js]: you were right about those &='s too. jshint caught them.
i'll do a unit test once we have all designs.

**Pull request merged**